client/java/src/main/java/io/openlineage/client/dataset/partition/trimmer/KeyValueTrimmer.java

```diff
@@ -0,0 +1,19 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.dataset.partition.trimmer;
+
+/** Normalizes if last part is a string represents a key value pair in an arbitrary format. */
+public class KeyValueTrimmer implements DatasetNameTrimmer {
+
+  private static final char EQUALITY_SIGN = '=';
+
+  @Override
+  public boolean canTrim(String name) {
+    String lastPart = getLastPart(name);
```

Comment:
IMHO this trimmer can detect the first part containing `=` symbol, and trim it with the rest of the string, instead of iterating over the same string multiple times. But current approach works as well